client/jetpack-cloud/sections/pricing/controller.tsx

```diff
@@ -25,11 +23,6 @@ export function jetpackPricingContext( context: PageJS.Context, next: () => void
 		return;
 	}
 
-	if ( /\/(pricing|plans)$/.test( pathname ) && siteFromUrl ) {
-		page.redirect( addQueryArgs( urlQueryArgs, `${ pathname }/${ siteFromUrl }` ) );
-		return;
-	}
-
```

Comment:
I'm removing this conditional redirect completely because the `/plans` route no longer exists, and there's no `/pricing/:site` route. A `/pricing?site=foo` path would redirect to `/pricing/foo` which doesn't exist.

Reply:
This creates a subtle difference between production and development.

If one of my sites owns a product and I visit `/pricing?site:site`, I get redirected to `/pricing/:site` which makes the product be highlighted as owned. Without the redirect, the product remains available.

Development (the redirect never happens and the state of the site is not reflected on the page)
![image](https://user-images.githubusercontent.com/3418513/125955186-ce091cd4-9042-4ae8-84be-fb7318c2abd9.png)

Production (the redirect does happen and the page shows that the site owns a product)
![image](https://user-images.githubusercontent.com/3418513/125955121-d096227d-2698-4039-b211-b768da1cfb81.png)

I guess that not having `/pricing/:site` defined as a route is a bug we introduced along the way. I can come back to this next Monday to check why and how things work without having a defined route for `/pricing/:site`.

Reply:
You're right, @rcanepa, the `/pricing/:site` route actually exists, and it was a mistake to remove the redirect conditional completely.

I re-added it in f91394991f, in a simplified form that accounts only for the `/pricing` routes, not `/plans`.

When you have a chance, please have another (final) look.

Reply:
Everything works as expected now, @jsnajdr. Thanks again for taking care of this.